src/services/upload/uploader.ts

```diff
@@ -0,0 +1,84 @@
+import { Collection } from 'services/collectionService';
+import { File } from 'services/fileService';
+import { sleep } from 'utils/common';
+import { handleError, CustomError } from 'utils/common/errorUtil';
+import { decryptFile } from 'utils/file';
+import { logError } from 'utils/sentry';
+import { fileAlreadyInCollection } from 'utils/upload';
+import NetworkClient from './networkClient';
+import UIService from './uiService';
+import { FileUploadResults } from './uploadManager';
+import UploadService, {
+    BackupedFile,
+    EncryptedFile,
+    FileInMemory,
+    UploadFile,
+} from './uploadService';
+
+const TwoSecondInMillSeconds = 2000;
+
+interface UploadResponse {
+    fileUploadResult: FileUploadResults;
+    file?: File;
+}
+export default async function uploader(
+    worker: any,
+    reader: FileReader,
+    existingFilesInCollection: File[],
+    rawFile: globalThis.File,
+    collection: Collection
+): Promise<UploadResponse> {
+    UIService.setFileProgress(rawFile.name, 0);
+    let file: FileInMemory = null;
+    let encryptedFile: EncryptedFile = null;
+    try {
+        file = await UploadService.readFile(reader, rawFile, collection);
+
+        if (fileAlreadyInCollection(existingFilesInCollection, file)) {
+            // set progress to -2 indicating that file upload was skipped
+            UIService.setFileProgress(rawFile.name, FileUploadResults.SKIPPED);
+            // wait two second before removing the file from the progress in file section
+            await sleep(TwoSecondInMillSeconds);
+            return { fileUploadResult: FileUploadResults.SKIPPED };
+        }
+
+        encryptedFile = await UploadService.encryptFile(
+            worker,
+            file,
+            collection.key
+        );
+
+        const backupedFile: BackupedFile = await UploadService.uploadToBucket(
+            encryptedFile.file
+        );
+
+        const uploadFile: UploadFile = UploadService.getUploadFile(
+            collection,
+            backupedFile,
+            encryptedFile.fileKey
+        );
+
+        const uploadedFile = await NetworkClient.uploadFile(uploadFile);
+        const decryptedFile = await decryptFile(uploadedFile, collection);
```

Comment:
What's the purpose of decrypting the file right after encrypting and uploading. Is it possible to reuse the `file` which is passed in the `UploadService.encryptFile`.  I am curious about the impact of this method call during upload of large video lines.

Reply:
So we store the decrypted file Key along with file in localDB  ,and we never receive the actual file Key back in the uploader function so, we have to decrypt it using the collection key. we could have avoided decrypting the metadata but already had a util function for decryption so used it. Decrpyting metadata is cheap operation

So we are not decrypting the actual file just the metadata of the file so it should not be a problem.

Reply:
Got it. I got confused. I was not aware that `uploadedFile` actually refers to `enteFile` object, not the media file.

Reply:
will make this change on different branch will change confusion here